scripts/controllers/main.js

```diff
@@ -43,14 +49,51 @@ angular.module('pbnApp')
   	  };
       };
 
-      $scope.addColor = function(color) {
-  	  $scope.palette.push(color);
+      $scope.colorDistance = function(c1, c2) {
+	// See https://stackoverflow.com/a/9085524
+	var r = c1.r - c2.r;
+	var g = c1.g - c2.g;
+	var b = c2.b - c2.b;
+	var rmean = r / 2;
+	var dist = Math.sqrt((((512+rmean)*r*r)>>8) + 4*g*g + (((767-rmean)*b*b)>>8));
+	return dist;
       };
 
+      $scope.addColor = function(color, threshold) {
+	  addColorInfo(color);
+	  var mindist = -1;
+  	  for (var i = 0; i < $scope.palette.length; i++) {
+               var pcol = $scope.palette[i];
+               if ((color.r == pcol.r) && (color.g == pcol.g) && (color.b = pcol.b)) {
```

Comment:
`==` not `=`

Reply:
Thanks for spotting this typo. I have fixed it in my repo.

Reply:
I ended up merging tour changes into my own fork, and fixed it there too: https://github.com/eric-wieser/pbnify